persistence-jdbc/core/src/main/scala/com/lightbend/lagom/internal/persistence/jdbc/SlickProvider.scala

```diff
@@ -38,12 +41,30 @@ private[lagom] class SlickProvider(
 
   val autoCreateTables: Boolean = createTables.getBoolean("auto")
 
+  private val asyncExecConfig =
+    new AsyncExecutorConfig(system.settings.config.getConfig("jdbc-defaults.slick.async-executor"))
+
   if (dbApi != null) {
     // Work around https://github.com/playframework/playframework/issues/7262
```

Comment:
This is fixed as of 2.6.0-RC1 so I think we can remove the workaround (which was never great because it assumes you haven't changed any of the relevant default config).